sandbox/visualizations_emp.R

```diff
@@ -11,176 +11,368 @@ library(leaflet.providers)
 
-# Define which test EXIF file to run the functions on
-#exif_file = exif_files[1]
+exif1$received_image_path <- extract_received_image_path(exif1)
 
-# Run for test EXIF file.
-#extract_metadata_emp(exif_file)
+exif2$received_image_path <- extract_received_image_path(exif2)
 
-# Run extraction on all EXIF files
-image_metadata <- purrr::map_dfr(exif_files, extract_metadata_emp)
+exif3$received_image_path <- extract_received_image_path(exif3)
 
+exif4$received_image_path <- extract_received_image_path(exif4)
 
-exif_file = exif_files[1]
+exif5$received_image_path <- extract_received_image_path(exif5)
 
-metadata1 <- ofo::extract_metadata_emp(exif_file)
+exif6$received_image_path <- extract_received_image_path(exif6)
 
+# now let's merge the exif files with the metadata files
 
+totalexif1 <- full_join(exif1, metadata1, by="received_image_path")
 
+totalexif2 <- full_join(exif2, metadata2, by="received_image_path")
 
+totalexif3 <- full_join(exif3, metadata3, by="received_image_path")
 
+totalexif4 <- full_join(exif4, metadata4, by="received_image_path")
 
+totalexif5 <- full_join(exif5, metadata5, by="received_image_path")
 
-#### Load data ####
+totalexif6 <- full_join(exif6, metadata6, by="received_image_path")
 
-## original data
+#### Now we're going to create an intelligible "time of collection" value that leaflet knows how to work with for each exif file ####
 
-exif_20220630_0041_original <- read.csv ("exif-examples\\exif_20220630-0041.csv")
+# creating new datetime column
 
-exif_20220730_0079_original <- read.csv ("exif-examples\\exif_20220730-0079.csv")
+totalexif1$datetime <- ymd_hms(totalexif1$DateTimeOriginal)
 
-exif_20230528_0008_original <- read.csv ("exif-examples\\exif_20230528-0008.csv")
+totalexif2$datetime <- ymd_hms(totalexif2$DateTimeOriginal)
 
-exif_20230528_0009_original <- read.csv ("exif-examples\\exif_20230528-0009.csv")
+totalexif3$datetime <- ymd_hms(totalexif3$DateTimeOriginal)
 
-exif_20230706_0152_original <- read.csv ("exif-examples\\exif_20230706-0152.csv")
+totalexif4$datetime <- ymd_hms(totalexif4$DateTimeOriginal)
 
-exif_20230706_0153_original <- read.csv ("exif-examples\\exif_20230706-0153.csv")
+totalexif5$datetime <- ymd_hms(totalexif5$DateTimeOriginal)
 
-## compiled image-level metadata. after loading each metadata file, add a row index ("X") identical to the row index in the original data files-- this will allow us to easily merge the two types of files in the next step.
+totalexif6$datetime <- ymd_hms(totalexif6$DateTimeOriginal)
 
-exif_20220630_0041_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20220630-0041.csv")
+# creating "duration in seconds from beginning of image to collection to time of [i] image" column. the number in brackets is the index position of the image that was collected first.
 
-exif_20220630_0041_metadata$X <- 1:nrow(exif_20220630_0041_metadata)
+totalexif1$duration <- as.duration(totalexif1$datetime[1] %--% totalexif1$datetime)
```

Comment:
Unclear why you can't just use `-` instead of %--%, but I'm probably missing something!

Reply:
No clue, I just followed instructions from the lubridate package documentation to the letter. The "-" doesn't do the same thing.

Reply:
I just left it as is, as it seems like no big deal, but let me know if you want me to dig into this more and try to change it